packages/block-editor/src/store/actions.js

```diff
@@ -771,7 +771,7 @@ export function* mergeBlocks( firstBlockClientId, secondBlockClientId ) {
 	const blockAType = getBlockType( blockA.name );
 
 	// Only focus the previous block if it's not mergeable
-	if ( ! blockAType.merge ) {
+	if ( ! blockAType?.merge ) {
```

Comment:
With the negation, we'll need to check if blockAType is truthy. We can accidentally reach this branch when blockAType is undefined:

<img width="234" alt="CleanShot 2021-09-16 at 15 05 49@2x" src="https://user-images.githubusercontent.com/1270189/133691875-33e7e435-49c7-4448-954a-083cbb2c6e3d.png">

So something like:

```jsx
if ( blockAType && ! blockAType.merge )
```

Reply:
Ooh yeah right, I have fixed it.